controllers/machineset_controller.go

```diff
@@ -631,11 +631,25 @@ func (r *MachineSetReconciler) calculateStatus(ctx context.Context, cluster *clu
 		}
 	}
 
-	newStatus.Replicas = int32(len(filteredMachines))
-	newStatus.FullyLabeledReplicas = int32(fullyLabeledReplicasCount)
-	newStatus.ReadyReplicas = int32(readyReplicasCount)
-	newStatus.AvailableReplicas = int32(availableReplicasCount)
-	return newStatus, nil
+	newMS.Status.Replicas = int32(len(filteredMachines))
+	newMS.Status.FullyLabeledReplicas = int32(fullyLabeledReplicasCount)
+	newMS.Status.ReadyReplicas = int32(readyReplicasCount)
+	newMS.Status.AvailableReplicas = int32(availableReplicasCount)
+	replicaConditionPresent := conditions.Get(newMS, clusterv1.ReplicasUpdated) != nil
+	if replicaCountsDifferent(&newMS.Status, &ms.Status) || !replicaConditionPresent {
+		// Delete and set the condition to force-update lastTransitionTime
+		replicasUpdatedCondition := clusterv1.CreateReplicasUpdatedCondition()
```

Comment:
It looks like this is used because the current `conditions.MarkTrue()` helper doesn't support setting the message and reason for the condition, I'm wondering if it would make sense to add a new helper to https://github.com/kubernetes-sigs/cluster-api/blob/master/util/conditions/setter.go for this use case rather than creating a one-off helper with `CreateReplicasUpdatedCondition()` here.

Reply:
I did this because I just felt it should be static and we should just update the time, since that's the only real element.  I think I prefer having the constructor pattern (maybe passing in custom message embedding counts as discussed below) rather than another generic helper.